code/game/objects/explosion.dm

```diff
@@ -1,94 +1,270 @@
+	// Calculate far explosion sound range. Only allow the sound effect for heavy/devastating explosions.
+	// 3/7/14 will calculate to 80 + 35
+	var/far_dist = 0
+	far_dist += heavy_impact_range * 5
+	far_dist += devastation_range * 20
+	var/frequency = get_rand_frequency()
+	for(var/mob/M in GLOB.player_list)
+		if(M.z == epicenter.z)
+			var/turf/M_turf = get_turf(M)
+			var/dist = get_dist(M_turf, epicenter)
+			// If inside the blast radius + world.view - 2
+			if(dist <= round(max_range + world.view - 2, 1))
+				M.playsound_local(epicenter, get_sfx("explosion"), 100, 1, frequency, falloff = 5) // get_sfx() is so that everyone gets the same sound
+			else if(dist <= far_dist)
+				var/far_volume = Clamp(far_dist, 30, 50) // Volume is based on explosion size and dist
+				far_volume += (dist <= far_dist * 0.5 ? 50 : 0) // add 50 volume if the mob is pretty close to the explosion
+				M.playsound_local(epicenter, 'sound/effects/explosionfar.ogg', far_volume, 1, frequency, falloff = 5)
 
-		var/took = (world.timeofday-start)/10
-		//You need to press the DebugGame verb to see these now....they were getting annoying and we've collected a fair bit of data. Just -test- changes  to explosion code using this please so we can compare
-		if(Debug2) to_world_log("## DEBUG: Explosion([x0],[y0],[z0])(d[devastation_range],h[heavy_impact_range],l[light_impact_range]): Took [took] seconds.")
+	if(adminlog)
+		log_and_message_admins("Explosion with size ([devastation_range], [heavy_impact_range], [light_impact_range]) in area [epicenter.loc.name] ([epicenter.x],[epicenter.y],[epicenter.z]) (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[epicenter.x];Y=[epicenter.y];Z=[epicenter.z]'>JMP</a>)")
 
-		sleep(8)
+	var/approximate_intensity = (devastation_range * 3) + (heavy_impact_range * 2) + light_impact_range
+	// Large enough explosion. For performance reasons, powernets will be rebuilt manually
+	if(!defer_powernet_rebuild && (approximate_intensity > 25))
+		defer_powernet_rebuild = 1
 
+	if(heavy_impact_range > 1)
+		var/datum/effect/system/explosion/E = new/datum/effect/system/explosion()
+		E.set_up(epicenter)
+		E.start()
+
+	var/x0 = epicenter.x
+	var/y0 = epicenter.y
+	var/z0 = epicenter.z
+	for(var/turf/T in trange(max_range, epicenter))
+		var/dist = sqrt((T.x - x0)**2 + (T.y - y0)**2)
+		if(dist < devastation_range)
+			dist = 1
+		else if(dist < heavy_impact_range)
+			dist = 2
+		else if(dist < light_impact_range)
+			dist = 3
+		else
+			continue
+
+		var/throw_dist = 9/dist
+		T.ex_act(dist)
+		if(!T)
+			T = locate(x0,y0,z0)
+		if(T)
+			var/throw_target = get_edge_target_turf(T, get_dir(epicenter,T))
+			for(var/atom_movable in T.contents)
+				var/atom/movable/AM = atom_movable
+				if(AM && AM.simulated && !T.protects_atom(AM))
+					AM.ex_act(dist)
+					if(!QDELETED(AM) && !AM.anchored)
+						addtimer(CALLBACK(AM, /atom/movable/.proc/throw_at, throw_target, throw_dist, throw_dist), 0)
```

Comment:
Throwing seems to be async anyways, so this should probably just be a regular call instead of a timer.

Reply:
As discussed on Discord, the timer seems to be needed for it to not cause GC problems.

Reply:
I still think that that should be changed to not use timers at some point, but that can be done in a later PR.